src/component/contents/DraftEditorBlock.react.js

```diff
@@ -83,50 +77,49 @@ class DraftEditorBlock extends React.Component {
    * will miss out on the browser natively scrolling to that position.
    *
    * To replicate native behavior, if the block overlaps the selection state
-   * on mount, force the scroll position. Check the scroll state of the scroll
-   * parent, and adjust it to align the entire block to the bottom of the
-   * scroll parent.
+   * on mount, force the scroll position. Check the visible bounds of the scroll
+   * parent against the bounds of the new block and scroll up or down to bring
+   * it into view.
    */
   componentDidMount(): void {
-    var selection = this.props.selection;
-    var endKey = selection.getEndKey();
+    const selection = this.props.selection;
+    const endKey = selection.getEndKey();
+
     if (!selection.getHasFocus() || endKey !== this.props.block.getKey()) {
       return;
     }
 
-    var blockNode = ReactDOM.findDOMNode(this);
-    var scrollParent = Style.getScrollParent(blockNode);
-    var scrollPosition = getScrollPosition(scrollParent);
-    var scrollDelta;
+    const blockElement = this._element;
+    if (blockElement == null) {
+      return;
+    }
+
+    const blockRect = blockElement.getBoundingClientRect();
+    const scrollParent = Style.getScrollParent(blockElement);
+    let scrollElement;
+    let scrollRect;
 
     if (scrollParent === window) {
-      var nodePosition = getElementPosition(blockNode);
-      var nodeBottom = nodePosition.y + nodePosition.height;
-      var viewportHeight = getViewportDimensions().height;
-      scrollDelta = nodeBottom - viewportHeight;
-      if (scrollDelta > 0) {
-        window.scrollTo(
-          scrollPosition.x,
-          scrollPosition.y + scrollDelta + SCROLL_BUFFER,
-        );
-      }
+      // Window itself has a different API for scrolling than elements especially on
+      // IE11.  Luckily we can get the documentElement (<HTML>) and operate on that.
+      scrollElement = window.document.documentElement;
```

Comment:
⭐️ ⭐️ ⭐️ Browsers don't agree on what the scrollable element is when the whole window scrolls. For some browsers, it's `<html>`, for others, it's `<body>` (see https://makandracards.com/makandra/55801-does-html-or-body-scroll-the-page).

We've addressed this in other parts of the code with a function that returns `document.scrollingElement` (which returns either `<body>` or `<html>` based on the browser implementation) and falls back to `<html>` for IE, which doesn't implement `document.scrollingElement`. https://github.com/textioHQ/frontend/blob/master/packages/webapp/app/es/core/utils/getPageScrollableElement.ts#L37-L40

Reply:
Great thanks! Yeah, I was going to use `scrollingElement`, but saw it wasn't available in IE11. When IE11 and Chrome agreed, I assumed I must have covered all possible cases 🤦‍♂.  I'll update that.